### DISCUSSION TITLE
How to create a component with wlc

### DISCUSSION THREAD

**Discussion by brianread108** [WeblateOrg/weblate#11992] [Q&A] [answered]
I am struggling to understand how I can create component from a zip file for an existing project with wlc or using curl from a batch file. When I do it on the web page it asks me all sorts of questions that I have to fill in. I can't see how to automate that using wlc. Can someone show me an example?

**Comment by nijel** [comment, marked as answer]
I don't think this is supported in wlc command line interface, but the API endpoint can do that: https://docs.weblate.org/en/latest/api.html#post--api-projects-(string-project)-components-

Just upload the file as `zipfile` and pass any parameters you want to configure.

**Comment by brianread108** [comment]
Is there a list of all the options for the upload?

**Reply by nijel** [reply]
Yes, it's linked from there: https://docs.weblate.org/en/latest/api.html#get--api-components-(string-project)-(string-component)-

**Comment by brianread108** [comment]
I've moved onto to trying to create the component from the gitea repo. Making progress. I shall be needing to automate this though using curl or wlc.

**Comment by brianread108** [comment]
Actually the Add Component Addon does the job for me, so no need for batch file to create the components.
